.circleci/config.yml

```diff
@@ -0,0 +1,159 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  environment:
+    LEIN_ROOT: "true"
+    JVM_OPTS: -Xmx3200m
+
+
+executors:
+  openjdk8:
+    docker:
+      - image: circleci/clojure:openjdk-8-lein-2.9.0
+    <<: *defaults
+  openjdk11:
+    docker:
+      - image: circleci/clojure:openjdk-11-lein-2.9.0
+    <<: *defaults
+
+jobs:
+  checkout_code:
+    executor: openjdk11
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v2-dependencies-{{ checksum "project.clj" }}
+            - v2-dependencies-
+      - run:
+          name: Downloading deps
+          command: |
+            lein deps
+      - save_cache:
```

Comment:
I would probably remove the `lein deps` call, and only save the cache right at the end of the build. Several of the jobs add dependencies which are downloaded later, and are then never cached. Also, some Leiningen plugins add dependencies dynamically, so they aren't actually downloaded until you run the actual command, so `lein deps` again doesn't pull everything that is actually required.

Reply:
Right. I know what you mean re the deps that never makes it into the cache. 

The more I think about it, the less useful the `checkout_code` job, as a standalone, is.